src/test/java/com/notably/logic/commands/suggestion/SearchSuggestionCommandTest.java

```diff
@@ -0,0 +1,148 @@
+        toCs2103Week1Lecture = AbsolutePath.fromString("/CS2103/Week1/Lecture");
+
+        // Set up model
+        BlockModel blockModel = new BlockModelImpl();
+        SuggestionModel suggestionModel = new SuggestionModelImpl();
+        ViewStateModel viewStateModel = new ViewStateModelImpl();
+        model = new ModelManager(blockModel, suggestionModel, viewStateModel);
+
+        // Add test data to model
+        Block cs2103 = new BlockImpl(new Title("CS2103"));
+        Block cs3230 = new BlockImpl(new Title("CS3230"));
+        model.addBlockToCurrentPath(cs2103);
+        model.addBlockToCurrentPath(cs3230);
+
+        Block week1 = new BlockImpl(new Title("Week1"));
+        Block week2 = new BlockImpl(new Title("Week2"));
+        Block week3 = new BlockImpl(new Title("Week3"));
+        model.setCurrentlyOpenBlock(toCs2103);
+        model.addBlockToCurrentPath(week1);
+        model.addBlockToCurrentPath(week2);
+        model.addBlockToCurrentPath(week3);
+
+        Block lecture = new BlockImpl(new Title("Lecture"));
+        model.setCurrentlyOpenBlock(toCs2103Week1);
+        model.addBlockToCurrentPath(lecture);
+
+        // Add body to the blocks
+        model.setCurrentlyOpenBlock(toCs2103Week1Lecture);
+        model.updateCurrentlyOpenBlockBody(toCs2103Week1LectureBody);
+
+        model.setCurrentlyOpenBlock(toCs2103Week2);
+        model.updateCurrentlyOpenBlockBody(toCs2103Week2Body);
+
+        model.setCurrentlyOpenBlock(toCs2103Week3);
+        model.updateCurrentlyOpenBlockBody(toCs2103Week3Body);
+
+        model.setCurrentlyOpenBlock(toCs3230);
+        model.updateCurrentlyOpenBlockBody(toCs3230Body);
+    }
+
+    @Test
+    public void constructor_nullPath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new SearchSuggestionCommand(null));
+    }
+
+    @Test
+    public void execute_nullModel_throwsNullPointerException() {
+        SearchSuggestionCommand searchSuggestionCommand = new SearchSuggestionCommand("keywords");
+        assertThrows(NullPointerException.class, () -> searchSuggestionCommand.execute(null));
+    }
+
+    @Test
+    public void execute_generateResponseCorrectly() {
+        SearchSuggestionCommand searchSuggestionCommand = new SearchSuggestionCommand("fail");
+        searchSuggestionCommand.execute(model);
+
+        // Test responseText
+        String expectedResponseText = "Search through all notes based on keywords";
+        assertEquals(Optional.of(expectedResponseText), model.responseTextProperty().getValue());
+
+        // Expected suggestions
+        List<SuggestionItem> expectedSuggestions = new ArrayList<>();
+        SuggestionItem suggestionItem = new SuggestionItemImpl(toCs3230.getStringRepresentation()
+                + WHITE_SPACE + "3", null);
+        SuggestionItem suggestionItem1 = new SuggestionItemImpl(toCs2103Week3.getStringRepresentation()
+                + WHITE_SPACE + "1", null);
```

Comment:
This is just FYI: You don't need to manually convert these numbers into string. Concatenating a string with a number will auto-convert the number into a string. That means, you can simply do `... + 3` and everything will work fine.